### ISSUE TITLE
Erroneous error note for using field after move with NLL enabled

### ISSUE THREAD

**Issue by carado** [rust-lang/rust#51512]
Hi, I noticed that the error for the following code:
```rust
#![feature(nll)]
fn main() {
    let range = 0 .. 1;
    let r = range;
    let x = range.start;
}
```
is this:
```
error[E0382]: use of moved value: `range.start`
 --> src/main.rs:5:13
  |
4 |     let r = range;
  |             ----- value moved here
5 |     let x = range.start;
  |             ^^^^^^^^^^^ value used here after move
  |
  = note: move occurs because `range.start` has type `i32`, which does not implement the `Copy` 
```
Notice the weird error note.

Without NLL enabled, the error becomes the more expected:
```
error[E0382]: use of moved value: `range.start`
 --> src/main.rs:4:9
  |
3 |     let r = range;
  |         - value moved here
4 |     let x = range.start;
  |         ^ value used here after move
  |
  = note: move occurs because `range` has type `std::ops::Range<i32>`, which does not implement the `Copy` trait
```
I'm running the latest nightly (as of this writing), which I got from rustup.
```
rustc 1.28.0-nightly (a805a2a5e 2018-06-10)
binary: rustc
commit-hash: a805a2a5ebba2802f432d79874e59c24e398f82a
commit-date: 2018-06-10
host: x86_64-unknown-linux-gnu
release: 1.28.0-nightly
LLVM version: 6.0
```
Thanks for your time.

**Comment by nikomatsakis** [comment]
The problem is that we are using `place` to get the path that we use in the note:

https://github.com/rust-lang/rust/blob/6ec1b626ba06b51fc8c23ee1cd7e2788163c2265/src/librustc_mir/borrow_check/error_reporting.rs#L128-L132

This corresponds to the place being *accessed* but not the place that was *moved*, which is what we actually want. The place(s) that were moved are noted in this loop here:

https://github.com/rust-lang/rust/blob/6ec1b626ba06b51fc8c23ee1cd7e2788163c2265/src/librustc_mir/borrow_check/error_reporting.rs#L84-L96

Here, the `moi` is a "move out index", which indexes into the `moves` vector give a `MoveOut` data structure:

https://github.com/rust-lang/rust/blob/6ec1b626ba06b51fc8c23ee1cd7e2788163c2265/src/librustc_mir/dataflow/move_paths/mod.rs#L174-L186

This contains the  path we want in its `path` field. That is a `MovePathIndex` which can be converted to a place through the `move_data`. We want to access the `move_paths` field:

https://github.com/rust-lang/rust/blob/6ec1b626ba06b51fc8c23ee1cd7e2788163c2265/src/librustc_mir/dataflow/move_paths/mod.rs#L124

and then use the field `place`:

https://github.com/rust-lang/rust/blob/6ec1b626ba06b51fc8c23ee1cd7e2788163c2265/src/librustc_mir/dataflow/move_paths/mod.rs#L93-L98

so something like `move_data.move_paths[mpi].place`.
